pkg/api/domain.go

```diff
@@ -41,3 +43,25 @@ func DomainForService(service Service) string {
 	}
 	return fmt.Sprintf("%s.%s", service, serviceDomain)
 }
+
+func DomainForContextAndService(context string, service Service) (string, error) {
```

Comment:
I don't think we should change anything in here, the only consume of this will remain that controller

Reply:
Thought we make a central place for domains, clusterNames, etc.
But I can do it later.